cashctrl_ledger/ledger.py

```diff
@@ -40,12 +40,40 @@ def vat_codes(self) -> pd.DataFrame:
         })
 
         return StandaloneLedger.standardize_vat_codes(result)
-    
+
     def mirror_vat_codes(self, target_state: pd.DataFrame, delete: bool = True):
         """
-        Not implemented yet
-        """
-        raise NotImplementedError
+        Aligns VAT rates on the remote CashCtrl account with the desired state provided as a DataFrame.
+
+        Parameters:
+            target_state (pd.DataFrame): DataFrame containing VAT rates in the pyledger.vat_codes format.
+            delete (bool, optional): If True, deletes VAT codes on the remote account that are not present in the target_state DataFrame.
+        """
+        current_state = self.vat_codes()
+        unique_entries = current_state.drop_duplicates(keep='first')
+        duplicates = current_state[current_state.index.duplicated(keep=False)]
```

Comment:
I believe the logic to work with duplicates does not work. 
`self.delete_vat_code(id)` removes every occurrence of id, so there is none left to update.

Here's a sounder logic for dealing with duplicates:
- if `delete=True` start by removing all duplicates
- if `delete=False` raise an Error

Reply:
Do you mean before finding entries to create/delete/update we need to delete all duplicated occurrences in the remote system, and only after that call the vat_codes() method?